src/context/json.ts

```diff
@@ -9,10 +17,23 @@ import { ResponseTransformer } from '../response'
  */
 export const json = <BodyType>(
   body: BodyType,
+  {
+    merge = defaultJSONContextOptions.merge,
+  }: JSONContextOptions = defaultJSONContextOptions,
 ): ResponseTransformer<BodyType> => {
   return (res) => {
     res.headers.set('Content-Type', 'application/json')
-    res.body = JSON.stringify(body) as any
+
+    if (merge) {
+      try {
+        const previousBody = JSON.parse(res.body as string)
```

Comment:
I think `BodyType` is typed wrong internally in the library from what I can tell. The `BodyType` is the generic of what is passed in as an e.g. object. It's then put as a JSON string on the `response`. The response typing however states that `res.body` is a `BodyType` while it is a stringified version of it. That's why there was the `as any` and we now need the `as string` to my understanding.

Reply:
I think you're right, the type may be wrong. Does it work with `res.body` typed as `string`?